### PR TITLE
Review persistence feature and tests

### PR DESCRIPTION
**Pull request by enyst**
Documentation:
- Add detailed Persistence section to agent-sdk-architecture.md in Layer 2: Runtime
- Document ConversationPersistence interface and FileStore implementation
- Include usage examples for basic usage, LocalConversation integration, and restoration
- Document file formats (events.jsonl and state.json), error handling, and custom implementations
- Add persistence support note to LocalConversation section

Tests:
- Add test for continuing conversation after restoration (full workflow)
- Add test for FileStore.listConversations() method
- Add test for corrupted state file handling
- Add test for corrupted event lines tolerance
- Add test for multiple conversations in same root directory
- Add tests for attachPersistence() on EventLog and ConversationState

The new tests cover previously untested functionality and edge cases, improving overall test coverage for the persistence feature.

### CODE REVIEW

File: packages/agent-sdk-ts/src/sdk/__tests__/persistence.test.ts [hunk 1]
```diff
@@ -98,4 +98,188 @@ describe('LocalConversation persistence', () => {
...
+    persistence1.appendEvent(event1);
+    persistence2.appendEvent(event2);
+
+    const events1 = persistence1.readEvents();
+    const events2 = persistence2.readEvents();
+
+    expect(events1).toHaveLength(1);
+    expect(events2).toHaveLength(1);
+    expect((events1[0] as any).llm_message.content[0].text).toBe('message A');
+    expect((events2[0] as any).llm_message.content[0].text).toBe('message B');
```
[the remark this answers is not in the document]
Comment (enyst) [reply packages/agent-sdk-ts/src/sdk/__tests__/persistence.test.ts]: /gemini wdyt now?

File: packages/agent-sdk-ts/src/sdk/__tests__/persistence.test.ts [hunk 2]
```diff
@@ -98,4 +98,188 @@ describe('LocalConversation persistence', () => {
...
+      llmClient: llm,
+      persistenceDir: dir,
+    });
+    restored.restoreConversation(id!);
+
+    const messagesBefore = (restored as unknown as { events: EventLog }).events.list().length;
+    await restored.sendUserMessage('second message');
+    const messagesAfter = (restored as unknown as { events: EventLog }).events.list().length;
+
+    expect(messagesAfter).toBeGreaterThan(messagesBefore);
```
[the remark this answers is not in the document]
Comment (enyst) [reply packages/agent-sdk-ts/src/sdk/__tests__/persistence.test.ts:138]: /gemini wdyt now?
